src/instructlab/config/init.py

```diff
@@ -125,6 +129,33 @@ def init(
         model_path = utils.expand_path(
             click.prompt("Path to your model", default=model_path)
         )
+
+    # check if model is present on disk or not
+    model_file = exists(model_path)
+    if model_file:
+        download_model_file = False
+    elif interactive:
+        download_model_file = click.confirm(
+            f"`{model_path}` seems to not exist. Should I download {DEFAULTS.GGUF_MODEL_NAME} to {DEFAULTS.MODELS_DIR} for you?"
```

Comment:
Why hardcode the default and not use `model_path` and `repository`?

Reply:
My thought was that a user could pass anything to those values - if something invalid is passed, we would just be issuing a download command that will fail anyway.

Since this is something that will only hit first-time users, it seemed appropriate to hold their hand a bit by just offering the defaults as a starting point.

That being said, I am not married to this approach, so I'd happily hear out other ideas!

Reply:
It's odd that if a user requests a specific model and that model isn't available, ilab suggests downloading a different one. How are users supposed to obtain the model they requested in this case?

Reply:
We could do a scheme where we attempt to download the requested model to the default model path (i.e. use `repository` and not `model_path`) though this will fail the `init` if the model they specified isn't one that exists.

Reply:
I think failing if the passed model is invalid makes sense. Thanks

Reply:
@leseb took another look at this - `repository` is just for the taxonomy repo, the only CLI arg for models is `model_path` currently - the only way I see us being able to use that to attempt a call to `ilab model download` is to try and truncate the path with something like `model_path.rsplit('/')` - I can go down this route but I feel that just using the default may be cleaner, wdyt?

Reply:
Not sure I follow, why truncate?

Reply:
Not sure how else to massage the `model_path` input into something more useful than just recommending the default

Reply:
After revisiting this, I think we should try to download whatever the user has requested, even though it might be incorrect. For this you can use:

```python
if ctx.get_parameter_source("model_path") == click.core.ParameterSource.COMMANDLINE
```

If the `model_path` was not passed by the CLI we can keep what you have suggested here. Otherwise we continue to propose to download what the user passed.

How does that sound?

Reply:
Sounds good to me!

Reply:
@leseb getting back to this now - this still is going to require the `model_path` massaging - I'm going to push up a rebased commit to show what I mean - no rush on this since we are targeting for `0.19.0`

Reply:
Okay I'm not even sure it is possible with the massaging - to get everything we need from `model_path` we'd have to extrapolate 3 different `ilab model download` args from the single `model_path` arg
```python
            ctx.invoke(
                download,
                repository=DEFAULTS.MERLINITE_GGUF_REPO,
                release="main",
                filename=DEFAULTS.GGUF_MODEL_NAME,
                model_dir=DEFAULTS.MODELS_DIR,
            )
```
Everywhere I have a default specified

Reply:
I can still do it for `model_dir` but I don't see how to do the others